csvy/readers.py

```diff
@@ -270,3 +270,30 @@ def read_to_list(
             data.append(row)
 
     return data, header
+
+
+def basic_read(
```

Comment:
This assumes that the first row contains the column names. Nothing wrong with this assumption, but it needs to be described in the docstring so the user knows what's going on.

Reply:
If the other changes are implemented, I'd rename this `read_into_dict` or something like that.